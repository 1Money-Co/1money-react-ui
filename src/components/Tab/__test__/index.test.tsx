import 'jsdom-global/register';
import * as React from 'react';
import { configure, shallow, render, mount } from 'enzyme';
import Adapter from '@cfaester/enzyme-adapter-react-18';
import sinon from 'sinon';
import { Tab } from '../index';

configure({ adapter: new Adapter() });

const originalConsoleError = console.error;
console.error = (message, ...optionalParams) => {
  if (
    message.includes('Could not parse CSS stylesheet') ||
    message.includes('findDOMNode is deprecated and will be removed')
  ) {
      return;
  }
  originalConsoleError(message, ...optionalParams);
};

describe('Tab', () => {
  it('renders correctly', () => {
    const wrapper = render(
      <Tab model={[
        { label: 'Password' },
        { label: '2FA settings' },
        { label: 'Limits' },
      ]} />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
