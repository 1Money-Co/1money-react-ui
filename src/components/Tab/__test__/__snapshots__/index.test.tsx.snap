// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tab renders correctly 1`] = `
LoadedCheerio {
  "0": <div
    class="usd1-react-ui-tab"
  >
    <div
      class="p-tabmenu p-component"
      data-pc-name="tabmenu"
      data-pc-section="root"
    >
      <ul
        class="p-tabmenu-nav p-reset"
        data-pc-section="menu"
        role="menubar"
      >
        <li
          class="p-tabmenuitem p-highlight"
          data-p-disabled="false"
          data-p-highlight="true"
          data-pc-section="menuitem"
          id="null_0"
          role="presentation"
        >
          <a
            aria-label="Password"
            class="p-menuitem-link"
            data-pc-section="action"
            href="#"
            role="menuitem"
            tabindex="0"
          >
            <span
              class="p-menuitem-text"
              data-pc-section="label"
            >
              Password
            </span>
          </a>
        </li>
        <li
          class="p-tabmenuitem"
          data-p-disabled="false"
          data-p-highlight="false"
          data-pc-section="menuitem"
          id="null_1"
          role="presentation"
        >
          <a
            aria-label="2FA settings"
            class="p-menuitem-link"
            data-pc-section="action"
            href="#"
            role="menuitem"
            tabindex="-1"
          >
            <span
              class="p-menuitem-text"
              data-pc-section="label"
            >
              2FA settings
            </span>
          </a>
        </li>
        <li
          class="p-tabmenuitem"
          data-p-disabled="false"
          data-p-highlight="false"
          data-pc-section="menuitem"
          id="null_2"
          role="presentation"
        >
          <a
            aria-label="Limits"
            class="p-menuitem-link"
            data-pc-section="action"
            href="#"
            role="menuitem"
            tabindex="-1"
          >
            <span
              class="p-menuitem-text"
              data-pc-section="label"
            >
              Limits
            </span>
          </a>
        </li>
        <li
          class="p-tabmenu-ink-bar"
          data-pc-section="inkbar"
          role="none"
        />
      </ul>
    </div>
  </div>,
  "_root": LoadedCheerio {
    "0": Document {
      "children": [
        <html>
          <head />
          <body />
        </html>,
      ],
      "endIndex": null,
      "next": null,
      "parent": null,
      "prev": null,
      "startIndex": null,
      "type": "root",
      "x-mode": "quirks",
    },
    "_root": [Circular],
    "length": 1,
    "options": {
      "decodeEntities": true,
      "xml": false,
    },
  },
  "length": 1,
  "options": {
    "decodeEntities": true,
    "xml": false,
  },
}
`;
